kale/loaddata/polypharmacy_datasets.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+import torch
+from torch.utils.data import Dataset
+from torch_geometric.data.data import Data
+from yacs.config import CfgNode
+
+from kale.utils.download import download_file_by_url
+
+
+class PolypharmacyDataset(Dataset):
+    r"""Polypharmacy side effect prediction dataset. Only for full-batch training.
+
+    Args:
+        cfg_dataset (CfgNode): configurations of the dataset.
+        mode (str): "train" or "test". Defaults to "train".
+    """
+
+    def __init__(self, cfg_dataset: CfgNode, mode: str = "train"):
+
+        super(PolypharmacyDataset, self).__init__()
+
+        self.cfg_dataset = cfg_dataset
+        data = self.load_data()
+
+        self.edge_index = data.__getitem__(f"{mode}_idx")
+        self.edge_type = data.__getitem__(f"{mode}_et")
+        self.edge_type_range = data.__getitem__(f"{mode}_range")
+
+        if mode == "train":
+            self.g_feat = data.g_feat
```

Comment:
can we use more descriptive variable names where, e.g., gene instead of g, have you decided to use gene or protein?

Reply:
likewise below for d